webrender/src/tiling.rs

```diff
@@ -2872,6 +2875,20 @@ impl FrameBuilder {
                                         device_length(self.screen_rect.size.height as f32,
                                                       device_pixel_ratio)));
 
+        // Pick a size for the cache render targets to be. The main requirement is that it
+        // has to be at least as large as the framebuffer size. This ensures that it will
+        // always be able to allocate the worst case render task (such as a clip mask that
+        // covers the entire screen).
+        // However, there are some extremely subtle rounding errors that occur in the
+        // reftests under OSMesa if the cache targets are exactly the size of the
+        // framebuffer. To work around this, we'll align the cache size to a multiple
+        // of the tile size. This can be removed once the tiling code is gone.
+        // TODO(gw): Remove this hack once the tiling code is sorted out!!
+        let max_dimension = cmp::max(screen_rect.size.width, screen_rect.size.height);
+        let aligned_max_dimension = (max_dimension + SCREEN_TILE_SIZE - 1) & !(SCREEN_TILE_SIZE-1);
+        let cache_size = DeviceUintSize::new(aligned_max_dimension as u32,
```

Comment:
if this code wasn't temporary, I'd ask for each of width/height to be rounded up separately